### PR TITLE
Fix strptime thread safety issue

### PR DESCRIPTION
**Pull request by leeclemens**
strptime thread safety hack-around - http://bugs.python.org/issue7980

### CODE REVIEW

**Comment by leeclemens** [comment]
Closes Issue https://github.com/fail2ban/fail2ban/issues/755

File: bin/fail2ban-client [hunk 1]
```diff
@@ -430,6 +430,9 @@ class ServerExecutionException(Exception):
 	pass
 
 if __name__ == "__main__": # pragma: no cover - can't test main
+	from time import strptime
+	# strptime thread safety hack-around - http://bugs.python.org/issue7980
+	strptime("2012", "%Y")
```
Comment (yarikoptic) [inline bin/fail2ban-client]: ha -- I think I had considered a similar hack at some point but don't remember why I haven't PRed it

shouldn't it be sufficient just to do once in ```fail2ban/__init__.py```.  that one I believe is imported by all the tools and we don't thread in the tools explicitely

**Comment by leeclemens** [comment]
My understanding is that it is best to call it as soon as possible - always before any other thread calls (possibly imports?) strptime - so I felt three separate calls in each of the __main__ sections made sense, otherwise I would agree a centralized common point of execution would be better.

**Comment by yarikoptic** [comment]
the irony is that they first import fail2ban, so ```__init__``` would take effect before running ```__main__``` ;-) so ```__init__.py``` will result in the eariiest point ;)

**Comment by leeclemens** [comment]
I misread what you said as putting it into `def __init__` for each, rather than `__init__.py`. I haven't experimented with that approach. Given the bug is fairly difficult to reproduce (race condition) and I have experienced this method consistently working, I chose to use it. If you think putting it into `__init__.py` would be better, I will happily move it.

**Comment by yarikoptic** [comment]
I do think that 1 place is better than 3 copies.
But I can't bloody recall why I haven't fixed it myself, although I remember looking into it... May be I just could not reproduce it and postponed

**Comment by leeclemens** [comment]
I think that should do it

**Comment by yarikoptic** [comment]
good.  changelog?  :-)

**Comment by leeclemens** [comment]
Kind of messy, want me to delete branch and re-submit as one commit? (Not sure how to merge after I pushed to my remote/origin)

**Comment by yarikoptic** [comment]
yeah -- that was a long way -- but it is the end what matters most ;)  cheers -- merging!

**Pull request merged**